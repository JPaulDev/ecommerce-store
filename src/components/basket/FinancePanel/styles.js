import styled from 'styled-components';

export const Container = styled.div`
  align-items: center;
  border-bottom: 1px solid var(--color-grey-9);
  column-gap: 25px;
  display: flex;
  padding: 30px 10px;

  img {
    height: auto;
    width: 100px;

    @media screen and (min-width: 600px) {
      width: 120px;
      height: auto;
    }
  }
`;

export const Text = styled.div`
  font-family: 'Uni Sans', sans-serif;
  font-weight: 600;
  line-height: 1.4;
`;

export const Link = styled.a`
  align-items: center;
  color: var(--color-blue-1);
  column-gap: 5px;
  display: flex;
  font-size: 0.75rem;
  font-weight: 600;
  margin-top: 10px;
  text-decoration: none;
  text-transform: uppercase;
`;
