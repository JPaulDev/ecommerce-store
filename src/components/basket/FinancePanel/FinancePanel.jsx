import Image from 'next/future/image';
import Link from 'next/link';
import { memo } from 'react';
import finance from '../../../../public/images/basket/finance.webp';
import paypal from '../../../../public/images/basket/paypal-credit.webp';
import { Info } from '../../icons';
import * as Styled from './styles';

function FinancePanel() {
  return (
    <div>
      <Styled.Container>
        <Image src={paypal} alt="PayPal credit." />
        <div>
          <Styled.Text>
            Spread the cost for up to 48 months (9.9% p.a fixed) or 0% for 4
            months.
          </Styled.Text>
          <Link href="/" passHref>
            <Styled.Link>
              <Info width={20} height={20} />
              More Info About PayPal Credit
            </Styled.Link>
          </Link>
        </div>
      </Styled.Container>
      <Styled.Container>
        <Image src={finance} alt="Pay on finance." />
        <div>
          <Styled.Text>
            Finance Classic Credit 48 Months 15.9%, Pay a deposit followed by 48
            monthly payments.
          </Styled.Text>
          <Link href="/" passHref>
            <Styled.Link>
              <Info width={20} height={20} />
              More Info About Finance Classic Credit
            </Styled.Link>
          </Link>
        </div>
      </Styled.Container>
    </div>
  );
}

export default memo(FinancePanel);
