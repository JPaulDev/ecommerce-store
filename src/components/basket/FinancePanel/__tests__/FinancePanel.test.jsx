import renderer from 'react-test-renderer';
import FinancePanel from '../index';

describe('FinancePanel component', () => {
  it('renders a panel with finance information and relevant links', () => {
    const tree = renderer.create(<FinancePanel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
