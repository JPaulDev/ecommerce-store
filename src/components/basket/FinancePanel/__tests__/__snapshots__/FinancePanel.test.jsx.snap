// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FinancePanel component renders a panel with finance information and relevant links 1`] = `
[
  <div
    className="sc-cad3a378-0 eSLspp"
  >
    <img
      alt="PayPal credit."
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
    <div>
      <div
        className="sc-cad3a378-1 hSHWKX"
      >
        Spread the cost for up to 48 months (9.9% p.a fixed) or 0% for 4 months.
      </div>
      <a
        className="sc-cad3a378-2 kWtZGW"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          fill="currentColor"
          height={20}
          viewBox="0 0 512 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
          />
        </svg>
        More Info About PayPal Credit
      </a>
    </div>
  </div>,
  <div
    className="sc-cad3a378-0 eSLspp"
  >
    <img
      alt="Pay on finance."
      data-nimg="future"
      decoding="async"
      height={24}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
      style={
        {
          "color": "transparent",
        }
      }
      width={24}
    />
    <div>
      <div
        className="sc-cad3a378-1 hSHWKX"
      >
        Finance Classic Credit 48 Months 15.9%, Pay a deposit followed by 48 monthly payments.
      </div>
      <a
        className="sc-cad3a378-2 kWtZGW"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          fill="currentColor"
          height={20}
          viewBox="0 0 512 512"
          width={20}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 512c141.4 0 256-114.6 256-256S397.4 0 256 0S0 114.6 0 256S114.6 512 256 512zM216 336h24V272H216c-13.3 0-24-10.7-24-24s10.7-24 24-24h48c13.3 0 24 10.7 24 24v88h8c13.3 0 24 10.7 24 24s-10.7 24-24 24H216c-13.3 0-24-10.7-24-24s10.7-24 24-24zm40-144c-17.7 0-32-14.3-32-32s14.3-32 32-32s32 14.3 32 32s-14.3 32-32 32z"
          />
        </svg>
        More Info About Finance Classic Credit
      </a>
    </div>
  </div>,
]
`;
